### PR TITLE
feat(components): add sideEffects flag for webpack and DCE

### PR DESCRIPTION
**Pull request by joshblack**
Quick enhancement for `carbon-components` by adding the `sideEffects` flag for tree-shaking and DCE in projects using webpack.

#### Changelog

**New**

**Changed**

- Add `sideEffects: false` to package.json

**Removed**

### CODE REVIEW

**Review by aledavila — APPROVED** [review]

**Review by asudoh — APPROVED** [review]
Though this comes with some risk I guess, but I wouldn't oppose.

**Comment by joshblack** [comment]
@asudoh happy to put the brakes on this 👀 would it help if we listed out explicit file paths that we knew didn't have side effects instead of specifying the whole package?

**Comment by asudoh** [comment]
> would it help if we listed out explicit file paths that we knew didn't have side effects instead of specifying the whole package?

If there is a way for us to tell WebPack what files don't have side effects, I see it'll be safer.

**Comment by joshblack** [comment]
@asudoh for sure! If I understand correctly, `sideEffects` can also take in an array of glob patterns we could specify for files that have side effects.

**Comment by asudoh** [comment]
> If I understand correctly, sideEffects can also take in an array of glob patterns we could specify for files that have side effects.

Thanks - In such case, the following files should be marked as _with_ side effects:

* [`es/bundle.js`](https://github.com/carbon-design-system/carbon/blob/v10.11.0/packages/components/src/bundle.js)
* [`es/globals/js/boot.js`](https://github.com/carbon-design-system/carbon/blob/v10.11.0/packages/components/src/bundle.js)
* [`es/globals/js/watch.js`](https://github.com/carbon-design-system/carbon/blob/v10.11.0/packages/components/src/globals/js/watch.js)

**Comment by joshblack** [comment]
Sounds good! Updated @asudoh

File: packages/components/package.json [hunk 1]
```diff
@@ -190,5 +190,10 @@
       "@babel/plugin-proposal-object-rest-spread",
       "dev-expression"
     ]
-  }
+  },
+  "sideEffects": [
+    "es/bundle.js",
+    "es/globals/js/boot.js",
+    "es/globals/js/watch.js"
```
Comment (asudoh) [inline packages/components/package.json]: Sorry for not telling `umd` directory:

```suggestion
    "es/bundle.js",
    "es/globals/js/boot.js",
    "es/globals/js/watch.js",
    "scripts/*",
    "umd/bundle.js",
    "umd/globals/js/boot.js",
    "umd/globals/js/watch.js"
```

**Comment by joshblack** [comment]
Updated!

**Review by tw15egan — APPROVED** [review]

**Pull request merged**